### PR TITLE
`Lifecycle.supportsDynamicLoad`

### PR DESCRIPTION
**Pull request by jglick**
In CloudBees CI running in high availability mode we run a custom lifecycle (#8589) which is not compatible with the option to dynamically install a plugin. While an individual plugin can be marked on the update center as requiring a controller restart to install, this was not available generically for _any_ plugin. `Lifecycle` seems like the right place for it since it already defined when and how to restart.

### Testing done

Interactively tested inside CloudBees CI by overriding this method.

### Proposed changelog entries

- N/A, beta-only API

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by Vlatombe — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Comment by MarkEWaite** [comment]
/label ready-for-merge

This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
